.github/workflows/build-docker-images.yml

```
@@ -68,27 +68,47 @@ jobs:
         run: |
           [[ "${REF}" == "master" ]] || git describe --tags --exact-match
 
-  build:
-    name: build docker image for a product
+  prepare-runner-os-list:
+    name: generate runner list
     runs-on: ubuntu-24.04
-    needs: validate-push-image
     outputs:
-      image-digest: ${{ steps.build-image.outputs.digest }}
+      runnerlist: ${{ steps.get-runnerlist.outputs.runnerlist }}
+    steps:
+      - run: sudo apt-get update && sudo apt-get -y install jo
+      - id: get-runnerlist
+        env:
+          INPUT_PLATFORMS: ${{ inputs.platforms }}"
+        run: |
+          runner_os_arch=()
+          echo $INPUT_PLATFORMS | grep -qq arm64 && runner_os_arch+=($(jo arch=linux/amd64 runner=ubuntu-24.04))
+          echo $INPUT_PLATFORMS | grep -qq arm64 && runner_os_arch+=($(jo arch=linux/arm64 runner=ubuntu-24.04-arm))
+          echo "runnerlist=$(jo -a ${runner_os_arch[@]})" >> "$GITHUB_OUTPUT"
+          # Fail if there are no runners for the requested architecture
```

architectures, plural, right? And we 'pass' as long as one requested architecture is present, instead of requiring completeness. Is that the right behaviour? ("yes" is a fine answer, in which case maybe add a comment explaining this)

---

While I was writing the answer, I realized that it would be cleaner to let it fail in the next job.

This made sense when we were using Ubicloud for `arm64` but as it is no longer the case, I will just remove this part (lines 86 - 89), and if `runnerlist` is empty, the next job will fail and the workflow will stop there

Thank you for your review!

---

makes sense!